system-design/margo-api-reference/workload-api/onboarding-api/rootca-download.md

```diff
@@ -0,0 +1,17 @@
+# Certificate API
+
+In order to facilitate secure communication between the device's management client and workload orchestration web service the workload orchestration web service's root CA certificate must be downloaded using the Onboarding API's `certificate` endpoint.
```

Comment:
I understand the need. However is there not a chicken and egg situation here. How do you securely download the cert without first having a cert to invoke the call?
I think we could benefit from some system flows around this area. Some questions spring to mind.
- Is it not the 'infrasture manager' (not yet defined) , who would prime a device with such plumbing info?
- Would be good to understand when such an onboarding service is invoked. It could be for devices that are unprovisioned and thus iteraction as part of this phase. It could be the devices are already provisioned (but they would need the stack to engage in flows) and need some hook to know when to trigger and against what destination.

So I wonder if this onboarding-api is mature enough to warrent inclusion in this PR.
would love to get an understanding of the macro system flows and state to add more value to this API.

Reply:
In order to ensure this PR can be pushed through to the pre-draft branch I will set this to resolved. However, this is a great point we should discuss in our next WOA focus group call and won't lose track of it.